src/ui/components/common/content_actions/post_content_actions.rs

```diff
@@ -0,0 +1,80 @@
+use crate::ui::components::common::content_actions::{ContentActionType, ContentActions};
+use lemmy_client::{
+  lemmy_api_common::{
+    lemmy_db_schema::newtypes::PostId,
+    lemmy_db_views::structs::PostView,
+    post::{CreatePostReport, PostReportResponse, PostResponse, SavePost as SavePostBody},
+  },
+  LemmyRequest,
+};
+use leptos::*;
+
+#[server(prefix = "/serverfn")]
+async fn save_post(id: PostId, save: bool) -> Result<PostResponse, ServerFnError> {
+  use crate::{constants::AUTH_COOKIE, utils::get_client_and_session};
+  let (client, session) = get_client_and_session().await?;
+
+  let jwt = session.get::<String>(AUTH_COOKIE)?;
+
+  client
+    .save_post(LemmyRequest {
+      body: SavePostBody { post_id: id, save },
+      jwt,
+    })
+    .await
+    .map_err(|e| ServerFnError::ServerError(e.to_string()))
+}
+
+#[server(prefix = "/serverfn")]
+async fn report_post(post_id: PostId, reason: String) -> Result<PostReportResponse, ServerFnError> {
```

Comment:
Just a note that it'd probably be easier to just pass the form objects in. That's what we do in jerboa for the most part.

These should also probably live in different files. IE separate your UI components from your API actions, IE


- a file for viewmodels / API actions / data persisted across different screens
- A file for the "screen" or main route with all the query-related stuff
- A file(s) for UI components.

Reply:
Agreed. I have similar, yet slightly different, ideas for organizing things. I'll open an issue detailing my ideas when I get the chance later today.

Reply:
Cool